log/logger.go

```diff
@@ -122,6 +184,71 @@ func init() {
 	globalMutex.Unlock()
 }
 
+func InitLoggerWithFile(cfg LogConfig) error {
+	err := checkLoggerConfigValid(cfg)
+	if err != nil {
+		return err
+	}
+	initDefaultLogger(cfg)
+	return nil
+}
+
+func InitTraceId(ctx *context.Context) {
+	if (*ctx).Value(ObkvTraceId) == nil {
+		pid := uint64(unix.Getpid())
+		// uniqueId uint64, pid uint64
+		uniqueId := uuid.New().String()
+		uniqueId = strings.ReplaceAll(uniqueId, "-", "")
+		traceId := fmt.Sprintf("Y%s-%x", uniqueId, pid)
+		*ctx = context.WithValue((*ctx), ObkvTraceId, traceId)
+	}
+}
+
+func checkLoggerConfigValid(cfg LogConfig) error {
+	if cfg.LogFileName == "" {
+		return errors.New("should set Log File Name in toml or client config")
+	} else if cfg.SingleFileMaxSize == 0 {
+		return errors.New("Single File MaxSize is invalid")
+	} else if cfg.SlowQueryThreshold == 0 {
+		return errors.New("Slow Query Threshold is invalid")
+	}
```

Comment:
how about other parameters？no need to check？

Reply:
down